server/src/test/java/org/elasticsearch/cluster/coordination/UnsafeBootstrapMasterIT.java

```diff
@@ -0,0 +1,236 @@
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.cluster.coordination;
+
+import joptsimple.OptionParser;
+import joptsimple.OptionSet;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.cli.MockTerminal;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.Manifest;
+import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.discovery.DiscoverySettings;
+import org.elasticsearch.discovery.zen.ElectMasterService;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.env.NodeEnvironment;
+import org.elasticsearch.env.NodeMetaData;
+import org.elasticsearch.env.TestEnvironment;
+import org.elasticsearch.node.Node;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.test.InternalTestCluster;
+import org.elasticsearch.test.junit.annotations.TestLogging;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.hamcrest.Matchers.containsString;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0, autoMinMasterNodes = false)
+@TestLogging("_root:DEBUG,org.elasticsearch.cluster.service:TRACE,org.elasticsearch.discovery.zen:TRACE")
+public class UnsafeBootstrapMasterIT extends ESIntegTestCase {
+
+    private int bootstrapNodeId;
+
+    @Before
+    public void resetBootstrapNodeId() {
+        bootstrapNodeId = -1;
+    }
+
+    @Override
+    protected List<Settings> addExtraClusterBootstrapSettings(List<Settings> allNodesSettings) {
+        if (internalCluster().size() + allNodesSettings.size() == bootstrapNodeId) {
+            List<String> nodeNames = new ArrayList<>();
+            Collections.addAll(nodeNames, internalCluster().getNodeNames());
+            allNodesSettings.forEach(settings -> nodeNames.add(Node.NODE_NAME_SETTING.get(settings)));
+
+            List<Settings> otherNodesSettings = allNodesSettings.subList(0, allNodesSettings.size() - 1);
+            Settings lastNodeSettings = allNodesSettings.get(allNodesSettings.size() - 1);
+            List<Settings> newSettings = new ArrayList<>();
+            newSettings.addAll(otherNodesSettings);
+            newSettings.add(Settings.builder().put(lastNodeSettings)
```

Comment:
Could we bootstrap a random node rather than it always being the last of them?

Reply:
https://github.com/elastic/elasticsearch/pull/37696/commits/f3d78c41dd0e9a155166ed21d34733b88a2c45a2